src/main/flight/mixer.c

```diff
@@ -661,6 +661,130 @@ void StopPwmAllMotors()
+
+    // consider rule only if no box assigned or box is active
+    if (!currentServoMixer[rule_index].box == 0 && !IS_RC_MODE_ACTIVE(BOXSERVO1 + currentServoMixer[rule_index].box - 1)) {
+        currentOutput[servo_index] = 0;
+        return rxConfig->midrc;
+    }
+
+    uint16_t servo_width = servoConf[servo_index].max - servoConf[servo_index].min;
+    int16_t min = currentServoMixer[rule_index].min * servo_width / 100 - servo_width / 2;
+    int16_t max = currentServoMixer[rule_index].max * servo_width / 100 - servo_width / 2;
+
+    bool needScale = true;
+
+    int16_t responseRange; //in range [-500:500] if needScale = true, else is [min:max]
+    int16_t input_from = currentServoMixer[rule_index].inputSource;
+
+    switch (input_from) {
+        case INPUT_STABILIZED_ROLL:
+        case INPUT_STABILIZED_PITCH:
+        case INPUT_STABILIZED_YAW:
+            //WARINING: this list of case works only if the order in rcData index enum is the same of inputSource_e
+            if (FLIGHT_MODE(PASSTHRU_MODE)) {
+                responseRange = rcCommand[input_from - INPUT_STABILIZED_ROLL];
+            } else {
+                responseRange = axisPID[input_from - INPUT_STABILIZED_ROLL];
+            }
+            // Reverse yaw servo when inverted in 3D mode
+            if (rule_index == INPUT_STABILIZED_YAW && feature(FEATURE_3D) && (rcData[THROTTLE] < rxConfig->midrc)) {
+                responseRange *= -1;
+            }
+            break;
+
+        case INPUT_STABILIZED_THROTTLE:
+            responseRange = motor[0] - 1000 - 500;  // Since it derives from rcCommand or mincommand and must be [-500:+500]
+            break;
+
+        case INPUT_RC_ROLL:
+        case INPUT_RC_PITCH:
+        case INPUT_RC_YAW:
+        case INPUT_RC_THROTTLE:
+        case INPUT_RC_AUX1:
+        case INPUT_RC_AUX2:
+        case INPUT_RC_AUX3:
+        case INPUT_RC_AUX4:
+            //WARINING: this list of case works only if the order in rcData index enum is the same of inputSource_e
+            responseRange = rcData[input_from - INPUT_RC_ROLL] - rxConfig->midrc;
+            break;
+
+        case INPUT_GIMBAL_PITCH:
+            needScale = false;
+            responseRange = decidegreeToServoPulse(servo_index, inclination.values.pitchDeciDegrees);
+            break;
+        case INPUT_GIMBAL_ROLL:
+            needScale = false;
+            responseRange = decidegreeToServoPulse(servo_index, inclination.values.rollDeciDegrees);
+            break;
+    }
+
+    //now, if the input, for coinvenience, has been mapped from [-500:500], we have to scale it to [min:max]
+    if (needScale){
+        if (responseRange > 0) {
+            responseRange = scaleRange(responseRange, 0, 500, 0, max);
+        } else {
+            responseRange = scaleRange(responseRange, 0, -500, 0, min);
+        }
+    }
+
+    if (currentServoMixer[rule_index].speed != 0){
+        if (currentOutput[rule_index] < responseRange)
```

Comment:
```
int step = responseRange - currentOutput[rule_index];
step = constrain(step, -currentServoMixer[rule_index].speed, currentServoMixer[rule_index].speed)
currentOutput[rule_index] += step;
```
And maybe even make step constraining conditional based on `currentServoMixer[rule_index].speed != 0` to avoid second code path